parlai/core/torch_classifier_agent.py

```diff
@@ -168,6 +170,127 @@ def value(self) -> float:
             return numer / denom
 
 
+class AUCMetrics(Metric):
+    """
+    Class that calculates the area under the roc curve from list of labels and its true
+    probabilities; expecting values to be (false positives, true positives)
+    """
+
+    __slots__ = (
+        '_pos_dict',
+        '_tot_pos',
+        '_neg_dict',
+        '_tot_neg',
+        '_class_name',
+        '_max_bucket_dec_places',
```

Comment:
I don't quite understand what this is for.

Reply:
According to a post I read.... adding properties/attributes to slots would make it faster to access.... but if it's not necessary, then I'll delete it.

Reply:
I meant more the specific property of `max_bucket_dec_places`